prefect_gcp/secret_manager.py

```diff
@@ -270,3 +284,91 @@ def example_cloud_storage_delete_secret_version_flow():
     partial_destroy = partial(client.destroy_secret_version, name=name, timeout=timeout)
     await to_thread.run_sync(partial_destroy)
     return name
+
+
+class SecretManager(SecretBlock):
+
+    gcp_credentials: GcpCredentials
+    secret_name: str = Field(default=..., description="Name of the secret to manage.")
+
+    @sync_compatible
+    async def read_secret(
+        self,
+        version_id: Union[str, int] = "latest",
```

Comment:
I think it makes sense to have the secret version be an attribute on the block, but still default to latest. Also I think we can just accept a string for the version. I can't think of any time that we'd need to do math with the version number, and we can cast it to an integer if that case does come up.